pkg/controller/worker/machines.go

```diff
@@ -238,10 +225,14 @@ func (w *workerDelegate) generateMachineConfig(ctx context.Context) error {
 
 		// VMO
 		if vmoDependency != nil {
+			workerPoolHash, err := w.generateWorkerPoolHash(pool, infrastructureStatus, vmoDependency, nil)
```

Comment:
The `workerPoolHash` is generated in 4 of 5 cases with same inputs. How about generating it once and override it in the `subnet != nil` case? That would improve in regards to DRY and eliminate at least one else case below.

Or even better in the `generateMachineClassAndDeployment()` function.